[WPE] Run MVT test suite

dpino opened the pull request:
#### c249bea4b3c612abc6f530469015863eb57f81a0
<pre>
[WPE] Run MVT test suite
<a href="https://bugs.webkit.org/show_bug.cgi?id=273679">https://bugs.webkit.org/show_bug.cgi?id=273679</a>

Reviewed by Carlos Alberto Lopez Perez.

Add script to run MVT tests.

Co-authored-by: Lauro Moura &lt;lmoura@igalia.com&gt;

* Tools/Scripts/libraries/resultsdbpy/resultsdbpy/__init__.py:
* Tools/Scripts/run-mvt-tests: Added.
(fatal):
(parse_args):
(options):
(suite_has_started):
(suite_has_finished):
(analyze_results):
(print_tests_failed):
(WPEWebDriver):
(WPEWebDriver.__init__):
(WPEWebDriver.start):
(WPEWebDriver.stop):
(WPEWebDriver._isRunning):
(WPEWebDriver._isFlatpakSDK):
(WPEWebDriver._isJHBuild):
(WPEWebDriver._run):
(WPEWebDriver._run.dict_flat):
(main):
* Tools/Scripts/webkitpy/__init__.py:

Canonical link: <a href="https://commits.webkit.org/281340@main">https://commits.webkit.org/281340@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/a4d3abde7a0de5e5bd4eb3a79c9d51e793a1474c

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/59604 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/38949 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/12128 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/63519 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/10127 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/61733 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/46602 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/10280 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/48362 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 wincairo-tests~~](https://ews-build.webkit.org/#/builders/60/builds/7094 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/61634 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/47/builds/36355 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/18/builds/51594 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🧪 api-wpe~~](https://ews-build.webkit.org/#/builders/41/builds/29200 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🧪 webkitpy](https://ews-build.webkit.org/#/builders/6/builds/59131 "Passed tests") | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/42/builds/33058 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/64/builds/8845 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/9051 "Built successfully") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/55005 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/63/builds/9123 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/65251 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/87/builds/3532 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/62/builds/9013 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/55701 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| [✅ 🧪 services](https://ews-build.webkit.org/#/builders/28/builds/59294 "Passed tests") | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/86/builds/3543 "Built successfully") | [  ~~🧪 mac-wk2-stress~~](https://ews-build.webkit.org/#/builders/8/builds/51586 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/55828 "Passed tests") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/88/builds/2931 "Passed tests") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/8900 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/34763 "Built successfully") | | | 
| | [  ~~🛠 tv-sim~~](https://ews-build.webkit.org/#/builders/46/builds/35846 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/36932 "Built successfully") | | | 
| | [  ~~🛠 watch-sim~~](https://ews-build.webkit.org/#/builders/45/builds/35591 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
<!--EWS-Status-Bubble-End-->

clopez commented on Tools/Scripts/run-mvt-tests:
```diff
@@ -0,0 +1,305 @@
+    print("OK")
+        
+    # 2. Start the WebDriver
+    args.browser = os.path.join("/home/dpino/workspace/gh/webkit", "WebKitBuild/WPE/Release/Tools/cog-prefix/src/cog-build/launcher/cog")
```
I think you mean top_level_directory instead of "/home/dpino/workspace/gh/webkit"

clopez replied:
Also why we override args.browser on main() ? It seems whatever browser path the user specifies via the arguments is used for nothing

clopez commented on Tools/Scripts/run-mvt-tests:
```diff
@@ -0,0 +1,305 @@
+            """,
+    )
+
+    parser.add_argument("address", help="Address of the remote WebDriver (i.e 0.0.0.0:8088)")
```
Instead of having the user to guess a port number here can we just automate it?
1. Pick one random port number between 1024 and 10000
2. Check that the port is free: https://stackoverflow.com/a/52872579
3. if is not free then repeat from 1

dpino replied:
In case the user doesn't set a port, the port 8088 is used by default:

https://github.com/WebKit/WebKit/pull/28091/files#diff-a86141502a4c4312c94880365b24458f8541a0c945bfe37e41fec15edc4d5fbcR103

Perhaps can make more clear that the port number is not mandatory.

clopez commented on Tools/Scripts/run-mvt-tests:
```diff
@@ -0,0 +1,305 @@
+    )
+    print("OK")
+
+    server = os.getenv("MVT_INSTANCE_ADDRESS", "https://mvt.rdkcentral.com")
```
why we do pass this via the environment instead of using an argument like the other parameters?

clopez commented on Tools/Scripts/run-mvt-tests:
```diff
@@ -0,0 +1,305 @@
+
+    # 7. Analyze results.
+    total, failed, passed = analyze_results('results.json')
+    print("Ran {} tests of which {} failed".format(total, failed))
```
Here I think it will be helpful also to print the names of the test that failed, so that get registered on the logs.
This will be helpful when checking later the buildbot logs we can see which tests failed on each run.

clopez reviewed (approved):
patch looks good, thanks, i added some comments to improve it a bit more.

dpino committed:
[WPE] Run MVT test suite
https://bugs.webkit.org/show_bug.cgi?id=273679

Reviewed by Carlos Alberto Lopez Perez.

Add script to run MVT tests.

Co-authored-by: Lauro Moura <lmoura@igalia.com>

* Tools/Scripts/libraries/resultsdbpy/resultsdbpy/__init__.py:
* Tools/Scripts/run-mvt-tests: Added.
(fatal):
(parse_args):
(options):
(suite_has_started):
(suite_has_finished):
(analyze_results):
(print_tests_failed):
(WPEWebDriver):
(WPEWebDriver.__init__):
(WPEWebDriver.start):
(WPEWebDriver.stop):
(WPEWebDriver._isRunning):
(WPEWebDriver._isFlatpakSDK):
(WPEWebDriver._isJHBuild):
(WPEWebDriver._run):
(WPEWebDriver._run.dict_flat):
(main):
* Tools/Scripts/webkitpy/__init__.py:

Canonical link: https://commits.webkit.org/281340@main
